tests/lib/openstack_query/managers/test_query_manager.py

```diff
@@ -129,23 +163,35 @@ def test_get_query_output_raises_error(self):
         """
         self.instance._get_query_output(MagicMock())
 
-    def test_populate_query_with_properties(self):
+    @parameterized.expand(
+        [
+            ("with all provided", MOCKED_OUTPUT_DETAILS_WITH_ALL),
+            ("with sort_by=None", MOCKED_OUTPUT_DETAILS_WITH_GROUP_BY),
+            ("with group_by=None", MOCKED_OUTPUT_DETAILS_WITH_SORT_BY),
+            ("with both as None", MOCKED_OUTPUT_DETAILS),
+        ]
+    )
+    def test_populate_output_params(self, _, mock_output_details):
         """
-        Tests that _populate_query method functions expectedly with properties
-        method builds the query with appropriate inputs before executing - calls select() with given properties
+        Tests that _populate_output_params method functions expectedly with output_details
+        method should set parameters related to parsing and formatting the output of the query
+        like sort, group and select - based on output_details given
         """
+        self.instance._populate_output_params(output_details=mock_output_details)
 
-        self.instance._populate_query(
-            MOCKED_OUTPUT_DETAILS.properties_to_select, MOCKED_PRESET_DETAILS
-        )
-        self.query.select.assert_called_once_with(
-            *MOCKED_OUTPUT_DETAILS.properties_to_select
-        )
-        self.query.where.assert_called_once_with(
-            preset=MOCKED_PRESET_DETAILS.preset,
-            prop=MOCKED_PRESET_DETAILS.prop,
-            **MOCKED_PRESET_DETAILS.args,
-        )
+        if mock_output_details.sort_by:
+            self.query.sort_by.assert_called_once_with(*mock_output_details.sort_by)
+        else:
+            self.query.sort_by.assert_not_called()
+
+        if mock_output_details.group_by:
```

Comment:
Ditto, there's three test cases here for each of the group_by / sort_by ...etc.
If there's common boilerplate extract it to a helper private method in the test. But tests like this tend to be very inflexible to future changes